From: Dan Carpenter <error27@gmail.com>
Date: Sat, 27 Aug 2011 12:59:30 +0300
Subject: mptfusion: unlock on error in mpt_config()
Git-commit: 83ff74e33849cd5592efa1ea583175600b6a485e
Patch-Mainline: 3.5
References: bnc#784805, FATE#313895

We introduced a new return here and forgot to unlock the mutex.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Acked-by: "Nandigama, Nagalakshmi" <Nagalakshmi.Nandigama@lsi.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/message/fusion/mptbase.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/message/fusion/mptbase.c b/drivers/message/fusion/mptbase.c
index 85c48ba..3489e72 100644
--- a/drivers/message/fusion/mptbase.c
+++ b/drivers/message/fusion/mptbase.c
@@ -6654,6 +6654,7 @@ mpt_config(MPT_ADAPTER *ioc, CONFIGPARMS *pCfg)
 				spin_unlock_irqrestore(&ioc->taskmgmt_lock, flags);
 				printk(MYIOC_s_INFO_FMT "%s: host reset in progress mpt_config timed out.!!\n",
 					__func__, ioc->name);
+				mutex_unlock(&ioc->mptbase_cmds.mutex);
 				return -EFAULT;
 			}
 			spin_unlock_irqrestore(&ioc->taskmgmt_lock, flags);
-- 
1.7.4.2

